### ISSUE TITLE
.

### ISSUE THREAD

**Issue by jodevak** [omoindrot/tensorflow-triplet-loss#25]

**Comment by omoindrot** [comment]
For the batch hard loss, you need to decrease the learning rate.

With the default learning rate of `1e-3`, the embeddings collapse to a single point so the loss is equal to the margin (`0.5`).

**Comment by jodevak** [comment]
Thank, but what about using it with Keras and a big number of classes? it would be a big help if it can !

**Comment by omoindrot** [comment]
I'm not sure what you are trying to do and what code you are using so some context would be helpful.

**Comment by jodevak** [comment]
Okay, im creating a model for image similarity. I already have a triplets dataset which are fed into my network in the form 3*batch_size. However while i'm using the above triplet loss (batch_all), the loss drops to 0 pretty quickly, which in turns mean, that i have easy triplet. Thus; i tried using the hard_triplet loss in this repository and the problem i have is that the loss doesn't get below the specified margin (e.g 0.5). I hope this explains. Thanks

**Comment by omoindrot** [comment]
The batch hard triplet loss in this repo works for batches with multiple images for each class.

For instance you would need to have a batch size of 100, which 10 images from each class.

In your case your triplets are already formed so you need to change the **triplet selection process**, not the triplet loss.

**Comment by jodevak** [comment]
Doesnt your function generates pairwise distances  based on labels ? i think its thus able to generate and select the hard samples, in other words i might consider my dataset as a seed dataset. Im a right ? thanks

**Comment by jodevak** [comment]
To summarize my request, can i extend you function to generate hard samples based on my seed dataset ? and if yes, what requirements must reside within the batch?
